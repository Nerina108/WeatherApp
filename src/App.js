//App.js - main component of the weather app, all sub components are called here to form the weather app.
//The weather app will ask the user to type in a city and country and it will return details on the weather conditions in that area

import React from 'react';
import './App.css';
import 'isomorphic-fetch';

//import sub-components
import Titles from './Components/titles';
import Form from './Components/form';
import Weather from './Components/weather';

//API key declared as constant
const Api_Key = "8d5b9c53633e02fb372a7bb8bd8a130f";

class App extends React.Component {

  //states declared as undefined
  state = {
    temperature: undefined,
    city: undefined,
    country: undefined,
    humidity: undefined,
    description: undefined,
    error: undefined
  }

  //promise used to get details
  getWeather = async (e) =>{

    //city and country declared as const
    const city = e.target.elements.city.value;
    const country = e.target.elements.country.value;
    //prevent errors
    e.preventDefault();
    //API call using fetch to access relevant information - set as const for easy access
    const api_call = await fetch(`http://api.openweathermap.org/data/2.5/weather?q=${city},${country}&appid=${Api_Key}`);
    const response = await api_call.json();
    console.log(response);

    //If both inputs are found, information is returned and stored.
    if (city && country){
      this.setState({
        temperature: response.main.temp,
        city: response.name,
        country: response.sys.country,
        humidity: response.main.humidity,
        description: response.weather[0].description,
        error: ""
      })
    } 
    //If either one or both inputs are not found, error message "Please input seach values is returned"
    else {
      this.setState({
        error: "Please input search values."
      })
    }
  }

  //render and return section calls all components and where necessary passes the relevant information.

  render(){
    return(
      <div className = "app">
        <div className= "main">
          <div className = "container">
            <div className = "row">
              <div className = "col-xs-5 title-container">
                <Titles />
              </div>
              <div className = "col-xs-7 form-container">
                <Form loadWeather={this.getWeather} />
                <Weather
                  temperature={this.state.temperature}
                  city={this.state.city}
                  country={this.state.country}
                  humidity={this.state.humidity}
                  description={this.state.description}
                  error={this.state.error}
                  />
              </div>
            </div>
          </div>
        </div>
      </div>
    )
  }
}

export default App;
