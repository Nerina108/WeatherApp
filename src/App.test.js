import React from 'react';
import renderer from 'react-test-renderer';
import App from './App';

//below the test method is is used to test the
//app component, it takes two values, the first
//value is the description(string), the second
//value is the actually logic for the test
test('Snapshot test for the component', () => {
   const tree = renderer
   .create(<App/>)
   .toJSON();
   expect(tree).toMatchSnapshot()
})
//test to check whether the fetch api return with an error
test('the fetch fails with an error', () => {
   return fetch('http://api.openweathermap.org/data/2.5/weather?q=Kimberley&APPID=67e35a09a8c57ed654d879b941a23b30')
   .catch(e => expect(e).not.toBe('error'));
 });

//test to check if the weather API is returning
//an object value and if it fails
test('API return should be string', () => {
   return fetch(`http://api.openweathermap.org/data/2.5/weather?q=Kimberley&APPID=67e35a09a8c57ed654d879b941a23b30`)
   .then(data => data.json())
   .then(dataReceived => {
       expect(typeof dataReceived).toBe('object')
   })
})

