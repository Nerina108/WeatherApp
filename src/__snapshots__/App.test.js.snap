// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot test for the component 1`] = `
<div
  className="app"
>
  <div
    className="main"
  >
    <div
      className="container"
    >
      <div
        className="row"
      >
        <div
          className="col-xs-5 title-container"
        >
          <div>
            <h2
              className="title-container_title"
            >
              Weather App
            </h2>
            <p
              className="title-container_subtitle"
            >
              Helps you find weather conditions in cities
            </p>
          </div>
        </div>
        <div
          className="col-xs-7 form-container"
        >
          <form
            onSubmit={[Function]}
          >
            <input
              name="city"
              placeholder="City"
              type="text"
            />
            <input
              name="country"
              placeholder="Country"
              type="text"
            />
            <button
              className="submit"
            >
              Get Weather
            </button>
          </form>
          <div
            className="weather"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;
